bin/p2-exec/main.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/square/p2/pkg/cgroups"
+	"github.com/square/p2/pkg/util"
+	"gopkg.in/yaml.v2"
+)
+
+// usage without cgroups:
+//   p2-exec username envdir launchable:cgname myexecutable myargs
+// there must be a PLATFORM_CONFIG_PATH envar (possibly in the envdir) to
+// specify the cgroup parameters for the <launchable>, which is created under
+// <cgname>
+func main() {
+	username := os.Args[1]
```

Comment:
mostly a matter of laziness and the fact that this program is really human-unfriendly anyway, but kingpin would make it easier to separate the launchablename and cgroupname so I'll implement that. Gonna cause a lot of churn in the PR though so review the rest of it and I'll amend the commits afterwards